### PR TITLE
feat: include initial authentication test cases

### PR DESCRIPTION
**Pull request by raimundo-henriques**

### CODE REVIEW

File: checklist.bs [hunk 1]
```diff
@@ -76,3 +76,72 @@ for the status code 202-case)
...
+Prior to this test, the test target host system must provide the testing party with the necessary
+credentials (i.e., a `client_id` and a `client_secret`).
+
+#### Request #### {#test-authenticate-success-request}
+
+An athentication request must be sent to the `AuthEndpoint` of the test target host system with
+**correct credentials** and the following syntax:
+
+<pre highlight=http>
+POST <a href="https://wbcsd.github.io/tr/data-exchange-protocol/#authendpoint">AuthEndpoint</a> HTTP/1.1
```
Comment (raimundo-henriques) [inline checklist.bs:99]: I believe this option might be overkill and hard to maintain. Please see the comment below for what I consider a better alternative.

File: checklist.bs [hunk 2]
```diff
@@ -76,3 +76,72 @@ for the status code 202-case)
...
+}
+```
+</div>
+
+### Test Case: Unsuccessful Authentication ### {#test-authenticate-unsuccessful}
+
+#### Request #### {#test-authenticate-unsuccessful-request}
+
+An authentication request must be sent to the `AuthEndpoint` of the test target host system with
+**incorrect credentials** and the syntax specified in [section 6.5.1 of the Technical Specifications](https://wbcsd.github.io/tr/data-exchange-protocol/#api-action-auth-request)
```
Comment (raimundo-henriques) [inline checklist.bs:128]: This would be my preferred alternative wrt linking the Tech Specs.

File: checklist.bs [hunk 3]
```diff
@@ -76,3 +76,360 @@ for the status code 202-case)
...
+Specifications](https://wbcsd.github.io/tr/data-exchange-protocol/#api-action-list-request). At that
+point, the third page of PCFs should be returned.
+
+The URL specified in the `link` of the response header of the second page must be stored. A
+ListFootprints GET request must be sent to this URL with a **valid access token** and the syntax
+specified in [section 6.6.3 of the Technical
+Specifications](https://wbcsd.github.io/tr/data-exchange-protocol/#api-action-list-request).
+
+#### Expected Response #### {#test-listfootprints-previous-response}
+
```
Comment (raimundo-henriques) [inline checklist.bs:430]: Given the expected response, I see no value in this as a test. I see value in communicating the possibility of links not being kept around, but not for a test case. @zeitgeist do you think we should keep it?

Comment (zeitgeist) [reply checklist.bs:496]: I've removed this test case

**Pull request merged**